graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/GeneratorHelper.java

```diff
@@ -38,26 +41,43 @@ private GeneratorHelper() {
    */
   public static Map<Byte, LongList> pickTopSocialProofs(
     SmallArrayBasedLongToDoubleMap[] socialProofs,
-    byte[] validSocialProofs,
     int maxSocialProofSize) {
 
     Map<Byte, LongList> results = new HashMap<>();
-    int length = validSocialProofs.length;
 
-    for (int i = 0; i < length; i++) {
-      SmallArrayBasedLongToDoubleMap socialProof = socialProofs[validSocialProofs[i]];
+    for (int i = 0; i < socialProofs.length; i++) {
+      SmallArrayBasedLongToDoubleMap socialProof = socialProofs[i];
       if (socialProof != null) {
         if (socialProof.size() > 1) {
           socialProof.sort();
         }
         socialProof.trim(maxSocialProofSize);
-        results.put(validSocialProofs[i], new LongArrayList(socialProof.keys()));
+        results.put((byte)i, new LongArrayList(socialProof.keys()));
```

Comment:
In the original implementation, we were passed a byte array of social proof types. Here we use the index as social proof types instead, but I think it is unsafe. For example, validSocialProofs could be {0, 2} whereas their indices are 0, 1, respectively. The new implementation will result in indexOutOfBound in the example given above, because it tries to access socialProofs based on indices, not social proof types. Why did we remove validSocialProofs in the first place?